klayout_package/python/kqcircuits/simulations/simulation.py

```diff
@@ -1455,3 +1458,81 @@ def visualise_region(
         else:
             for i, p in enumerate(points):
                 self.cell.shapes(visualisation_layer).insert(pya.DText(f"{label}_{i+1}", p.x, p.y))
+
+    def visualise_ports(self, edge_port_thickness=500):
+        """Visualise all ports in a dedicated 'simulation_ports' layer using get_port_data().
+
+        Args:
+            edge_port_thickness (float): Extra outward extension (µm) for EdgePorts to make them more visible.
+        """
+        dbu = self.layout.dbu
+        port_json = self.get_port_data()
+
+        for port in self.ports:
+            # Initialize visualise_point at the start of each iteration
+            visualise_point = None
+
+            # Try to find a matching entry in port_json
+            port_data_list = [p for p in port_json if p.get("number") == port.number]
+            if not port_data_list:
+                logging.warning(f"Port {port.number} not found in get_port_data() output, skipping visualisation.")
+                continue
+            port_data = port_data_list[0]
+
+            # Label based on port type
+            if isinstance(port, EdgePort):
+                label = f"edge_port_{port.number}"
+            elif isinstance(port, InternalPort):
+                label = f"internal_port_{port.number}"
+            else:
+                logging.warning(
+                    f"Unsupported port type {type(port).__name__} for port {port.number}"
+                )
```

Comment:
```suggestion
                logging.warning(f"Unsupported port type {type(port).__name__} for port {port.number}")
```
Sorry for having these endless tweaks, I promise that this set of suggestions is the last one :D Running the black tool wants this change, just click apply suggestion